src/main/java/com/epam/learn/springcore/service/TraineeService.java

```diff
@@ -1,67 +1,141 @@
 package com.epam.learn.springcore.service;
 
-import com.epam.learn.springcore.dao.TraineeDAO;
+import com.epam.learn.springcore.dao.TraineeRepository;
+import com.epam.learn.springcore.dao.UserRepository;
 import com.epam.learn.springcore.entity.Trainee;
+import com.epam.learn.springcore.entity.Trainer;
+import com.epam.learn.springcore.entity.Training;
+import com.epam.learn.springcore.entity.User;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
 import lombok.extern.log4j.Log4j2;
-import org.apache.commons.lang3.RandomStringUtils;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDate;
+import java.util.List;
+
 @Service
 @Log4j2
 public class TraineeService {
-    @Autowired
-    private TraineeDAO traineeDAO;
+    private final TraineeRepository traineeRepository;
+    private final UserService userService;
+
+    public TraineeService(TraineeRepository traineeRepository, UserService userService) {
```

Comment:
To avoid voilarplate code add @RequiredArgsConstructor.

Reply:
Ok. I'll use it for the next task.